projects/sequence_prediction/continuous_sequence/run_knn.py

```diff
@@ -184,22 +190,35 @@ def saveResultToFile(dataSet, predictedInput):
 
   k = int(np.floor(numTrain/24)-1)
 
-  predictedInput = np.zeros((len(sequence),))
+
   targetInput = np.zeros((len(sequence),))
-  trueData = np.zeros((len(sequence),))
+  predictedInput = np.zeros((len(sequence),))
+  predictedInput2 = np.zeros((len(sequence),))
   for i in xrange(numTrain, len(sequence) - predictionStep):
     testInstance = sequence[i + predictionStep]
-    neighbors = getNeighbors(sequence[:i], testInstance, k, nFeature)
-
     targetInput[i] = testInstance[-1]
+
+    # select data points at that shares same timeOfDay and dayOfWeek
+    neighbors = getNeighbors(sequence[:i], testInstance, k, [0, 1])
     predictedInput[i] = getResponse(neighbors)
 
-    print "step %d, target input: %d  predicted Input: %d " % (
-      i, targetInput[i], predictedInput[i])
+    # run KNN within the selected data points using passengerCount as feature
+    neighbors2 = getNeighbors(neighbors, testInstance, 10, [2])
+
+    weights = abs(np.array(neighbors2)[:, 2] - testInstance[2]) + 0.01
```

Comment:
Good catch! I have corrected the weighting function. It doesn't seem to affect the performance much in this case though.